### ISSUE TITLE
Bank account - Able to proceed with invalid information in company information

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#14107]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Login with any account
2. Open Settings => Workspace => Connect Bank Account => Connect Manually => Enter valid information for first 3 steps.
3. Press back go to step 2, now enter invalid information for phone number (or Zipcode, Tax Id ), after that press X to close.
4. Go back to Settings => Workspace => Connect Bank Account => Continue with setup, now you are at step 3.  Try to press Save & Continue
5. Notice that it still let you go to the next step.
6. From Scan document screen, click X  to go to previous steps, click back again to go to steps 2.
7. Notice that at the steps 2 it's invalid phone number.

## Expected Result:
After pressing Continue with setup, it should navigate to the invalid screen information.

## Actual Result:
You still can go to Scan Document Screen with invalid information.

## Workaround:
 Only save user information (Steps 2,3) when user press Save or we need to navigate user to the invalid information screen after press Continue with setup

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] iOS / native

**Version Number:** 1.2.50-2
**Reproducible in staging?:** y 
**Reproducible in production?:** Could not check
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/211127819-c840bfcf-1dc3-4d29-8339-60a5a5f249f7.MP4

https://user-images.githubusercontent.com/43996225/211127844-340ccc06-c2fd-4848-a0f4-84a5e599e531.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @hungvu193 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1673019074038549

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by yuwenmemon** [comment]
Will take a look at this one!

**Comment by kadiealexander** [comment]
Sorry, I was OOO when this was assigned to me 😬. I'm running into the following error when trying to replicate this, are you able to replicate it @yuwenmemon or should I take it to the team to see if someone else can?

**Comment by kadiealexander** [comment]
Not overdue.

**Comment by yuwenmemon** [comment]
As pointed out [here](https://expensify.slack.com/archives/C049HHMV9SM/p1673344972597009?thread_ts=1673019074.038549&cid=C049HHMV9SM) and observed first hand, we're not actually _saving_ the invalid information when you close out the form with it. So I guess the real issue is here that we're not calling to attention the invalid form field immediately when you press the back button? However at the end of the day that's a _pretty_ small bug...

**Comment by yuwenmemon** [comment]
Curious what @marcaaron and @luacmartins since they worked on the [Forms Design Doc](https://docs.google.com/document/d/1DI4jQ0nyfi2JczhPT440orwlXsTP_O-aGvg-HLmcbwI/edit#heading=h.ns12dr1f35rx) - should we consider this a bug? That an invalid _draft_ value is not immediately flagged as invalid?

**Comment by kadiealexander** [comment]
Not overdue.

**Comment by luacmartins** [comment]
As pointed out, we just showing a draft value that hasn't been saved yet. I think this follows our form validation guidelines and is not a bug.

**Comment by marcaaron** [comment]
I don't think this is a bug so gonna close. We don't validate the draft values when a form opens just when you blur a field or submit the form. If we want to change the way that works then it's more of a feature request than a bug. I don't really see much value in doing that though. Feel free to reopen if anyone disagrees!
